### ISSUE TITLE
Shuttles performance

### ISSUE THREAD

**Issue by justinanderson** [MIT-Mobile/MIT-Mobile-for-iOS#834]
Got our first report from an iPhone 4 user that Shuttles is a lot slower for them in 4.0 than the last release.

1. Takes about 30 seconds from a fresh launch for the Shuttles main screen to show anything but the refresh control, regardless of whether the user has Location Services on or not.
2. Take about 30 more seconds from tapping Tech Shuttle to its detail screen updating predictions for stops. One user reported that it looked like the app was waiting for the map to finish drawing before fetching predictions. My guess is that it's just taking that long for predictions to be parsed.

Guesses as to how we can speed things up:

- Upgrade to RestKit 0.24.0. It claims performance improvements with nested managed objects over earlier versions. I just quickly tried it myself and didn't see any performance improvement, but I might have not done it correctly.

- Reduce the number of relationships in the Core Data model. It's possible that parsing and fetching is taking so long because Shuttles has such a complex data model.

- Index the Core Data attributes we search by. Right now nothing's indexed, and that might have an effect on fetch speed.

- Instead of loading and parsing MITShuttlesDefaultRoutesData.json to prefill Core Data before the web request has finished, build the app with a prefilled persistent store. This likely won't make fresh data load any faster, but that initial shot of possibly stale cached data would ideally be an instant fetch, so the user has routes to pick from if they want to drill into a specific route before the web requests have finished loading.

**Comment by rosslebeau** [comment]
Cool, I've already been investigating and pruning the Core Data model a bit. I'll do performance checks on various things and see what happens. I can probably also get an iPhone 4 to test on.

**Comment by justinanderson** [comment]
I've got a 4 I can lend you if you can't find one.

**Comment by rosslebeau** [comment]
@justinanderson Just to give you an update - this is what I'm currently working on. I updated RestKit and measured the improvements; we saved about 0.5 seconds on the iPad I'm testing on, which normally takes ~10 seconds to load. Most of the time (9+ seconds) is spent fulfilling NSFetchRequests, so I'm digging into the app with various Instruments tools to see if I can find out where these requests are coming from and how to reduce them, or their complexity.
